src/transformers/generation/configuration_utils.py

```diff
@@ -382,6 +384,7 @@ def __init__(self, **kwargs):
 
         # Cache implementation
         self.cache_implementation = kwargs.pop("cache_implementation", None)
+        self.return_legacy_cache = kwargs.pop("cache_implementation", True)
```

Comment:
bit weird to fetch `cache_implementation`

Reply:
Indeed, should be `return_legacy_cache`, this one is clearly my mistake!